lib/utils/testlog/log.go

```diff
@@ -42,5 +42,64 @@ func FailureOnly(t *testing.T) *logrus.Entry {
 		fmt.Fprintln(os.Stderr, buf.String())
 	})
 
-	return logrus.NewEntry(log)
+	return utils.WrapLogger(logger.WithField("test", t.Name()))
+}
+
+// NewCheckTestWrapper creates a new logging wrapper for the specified
+// instance of the gocheck.C.
+// Returned value has an exported Log attribute that represents
+// the logger for the underlying test.
+// It is caller's responsibility to release the wrapper by invoking
+// Close after the test has completed.
+func NewCheckTestWrapper(c *check.C) *TestWrapper {
+	w := &TestWrapper{
+		c: c,
+	}
+	w.Log = FailureOnly(w)
+	return w
+}
+
+// Cleanup registers the specified handler f to be run
+// after the hest gas completed
```

Comment:
```suggestion
// after the test has completed
```

Reply:
Addressed in https://github.com/gravitational/teleport/pull/4849/commits/756c17b66db83fc5637415d0231fd0324cd72519